### PR TITLE
fix: [CDS-71679]: fixed selective stage popover calculation and also the scroll behaviour

### PR DESCRIPTION
**Pull request by Sagar-Rajput7**
### Summary

| Before  | After  |
|:-:|---|
| image  | image  |

**Scroll behaviour**

Before: 

https://github.com/harness/harness-core-ui/assets/98325173/4f4252e5-d1bf-449b-b637-d82ccd15dbe9

After: 

https://github.com/harness/harness-core-ui/assets/98325173/cee1bd42-d777-4ec5-b89a-994b5e582c6f

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionMetadata/ExecutionMetadata.tsx [hunk 1]
```diff
@@ -148,22 +148,25 @@ export default function ExecutionMetadata(): React.ReactElement {
 
   const renderSingleStageExecutionInfo = (): React.ReactElement | null => {
     const countLen = defaultTo(pipelineExecutionSummary?.stagesExecuted?.length, 0)
+    const shouldShowPopover = countLen - LIMIT > 0
```
Comment (rameshx) [inline src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionMetadata/ExecutionMetadata.tsx:151]: `countLen` can be renamed to `stagesExecutedCount` or `stageCount`

Comment (rameshx) [inline src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionMetadata/ExecutionMetadata.tsx:151]: easy to read/understand
```suggestion
    const shouldShowPopover = countLen > LIMIT
```

Comment (Sagar-Rajput7) [reply src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionMetadata/ExecutionMetadata.tsx]: done

**Review by rameshx — APPROVED** [review]

**Review by ishantmehtaharness — APPROVED** [review]
looks good

**Pull request merged**